packages/slate/src/changes/with-schema.js

```diff
@@ -121,12 +104,22 @@ function normalizeNodeAndChildren(change, node, schema) {
  * @return {Node}
  */
 
-function refindNode(change, node) {
+function refindNode(change, node, lastPath) {
   const { value } = change
   const { document } = value
-  return node.object == 'document'
-    ? document
-    : document.getDescendant(node.key)
+  if (node.object === 'document') {
+    return { node: document, nodePath: lastPath }
+  }
+  const tryNode = document.getDescendantAtPath(lastPath)
+  if (tryNode && tryNode.key === node.key) {
+    return { node: tryNode, nodePath: lastPath }
+  }
+  node = document.getDescendant(node.key)
+  if (!node) {
+    return { node, nodePath: [] }
+  }
+  const nodePath = document.getPath(node.key)
+  return { node, nodePath }
```

Comment:
Can you change these return values to `{ node, path }`?

Reply:
Yes, I will.